source/authentication/apple.txt

```diff
@@ -59,379 +59,607 @@ following:
-      private key through the Apple Developer Portal.
+            You can now create the client secret JWT for the Apple ID
+            authentication provider. Make sure that you have the following
+            information:
 
-      1. Click :guilabel:`Keys` in the left navigation menu.
+            - The :guilabel:`Bundle ID` of your app.
 
-      #. Click the blue plus icon next to :guilabel:`Keys`.
+            - The :guilabel:`Key ID` of the key that you created and the ``.p8``
+               file that contains the key.
 
-      #. On the :guilabel:`Register a New Key` page, enter a descriptive
-         :guilabel:`Key Name` and then scroll down to find the
-         :guilabel:`Sign in with Apple` row. Check the checkbox to enable
-         Sign in with Apple and then click :guilabel:`Configure`.
+            - Your Apple Team ID. You can find this in the top right of the Apple
+               Developer Portal.
 
-      #. On the :guilabel:`Configure Key` page, select the App ID that you
-         created as the :guilabel:`Primary App ID` and then click
-         :guilabel:`Save`.
+            .. figure:: /images/apple-team-id.png
+               :alt: An Apple team ID in the Apple Developer Portal
 
-         .. figure:: /images/apple-auth-configure-key.png
-            :alt: The Configure Key page in the Apple Developer Portal
+            Once you've confirmed that you have all the required information, you can use
+            a script to generate the JWT. You may define your own script or use the
+            script in this step.
 
-      #. Click :guilabel:`Continue` to review your key configuration. When
-         you're sure that you've configured the key correctly, click
-         :guilabel:`Register`.
+            To generate the JWT, we'll use the :rubygems:`jwt <gems/jwt>` gem. To install
+            it, run the following:
 
-      #. Copy the :guilabel:`Key ID` value somewhere that you can access it
-         later and then click :guilabel:`Download` to download the key as a
-         ``.p8`` text file. You will use these to generate the client
-         secret.
+            .. code-block:: shell
+
+               gem install jwt
 
-         .. important:: Save the Key
+            Create a new file called ``generate_client_secret.rb`` and copy the following
+            code block into the file.
 
-            You can only download the key one time. Make sure that you save
-            the key somewhere safe in case you need it again. If you lose
-            the key, you will need to generate a new one.
+            .. code-block:: ruby
 
+               # Source: https://developer.okta.com/blog/2019/06/04/what-the-heck-is-sign-in-with-apple
+               require 'jwt'
 
-   .. step:: Create the Client Secret JWT
+               # Update these values with your app's information
+               team_id = '<Apple Team ID>'
+               client_id = '<Bundle ID>'
```

Comment:
Can we explicitly indicate somewhere that `client_id` is the Bundle ID for mobile apps? Preferably outside the code block so that we can reuse the code for web.